apps/api/v1/pages/api/connected-calendars/_get.ts

```diff
@@ -0,0 +1,146 @@
+ *                     type: string
+ *                   appId:
+ *                     type: string
+ *                   userId:
+ *                     type: number
+ *                   integration:
+ *                     type: string
+ *                   calendars:
+ *                     type: array
+ *                     items:
+ *                       type: object
+ *                       properties:
+ *                         externalId:
+ *                           type: string
+ *                         name:
+ *                           type: string
+ *                         primary:
+ *                           type: boolean
+ *                         readOnly:
+ *                           type: boolean
+ *             examples:
+ *               connectedCalendarExample:
+ *                 value: [
+ *                   {
+ *                     "name": "Google Calendar",
+ *                     "appId": "google-calendar",
+ *                     "userId": 10,
+ *                     "integration": "google_calendar",
+ *                     "calendars": [
+ *                       {
+ *                         "externalId": "alice@gmail.com",
+ *                         "name": "alice@gmail.com",
+ *                         "primary": true,
+ *                         "readOnly": false
+ *                       },
+ *                       {
+ *                         "externalId": "addressbook#contacts@group.v.calendar.google.com",
+ *                         "name": "birthdays",
+ *                         "primary": false,
+ *                         "readOnly": true
+ *                       },
+ *                       {
+ *                         "externalId": "en.latvian#holiday@group.v.calendar.google.com",
+ *                         "name": "Holidays in Narnia",
+ *                         "primary": false,
+ *                         "readOnly": true
+ *                       }
+ *                     ]
+ *                   }
+ *                 ]
+ *       401:
+ *        description: Authorization information is missing or invalid.
+ *       403:
+ *        description: Non admin user trying to fetch other user's connected calendars.
+ */
+
+async function getHandler(req: NextApiRequest) {
+  const { userId, isAdmin } = req;
+
+  if (!isAdmin && req.query.userId) throw new HttpError({ statusCode: 403, message: "ADMIN required" });
+
+  const userIds = req.query.userId ? extractUserIdsFromQuery(req) : [userId];
+
+  const usersWithCalendars = await prisma.user.findMany({
+    where: { id: { in: userIds } },
+    include: {
+      selectedCalendars: true,
+      destinationCalendar: true,
+    },
+  });
+
+  return await getConnectedCalendars(usersWithCalendars);
+}
+
+async function getConnectedCalendars(users: UserWithCalendars[]) {
+  const connectedDestinationCalendarsPromises = users.map((user) =>
+    getConnectedDestinationCalendars(user, false, prisma).then((connectedCalendarsResult) =>
```

Comment:
NIT:
I would rename the `connectedDestinationCalendarsPromises` to `connectedAndDestinationCalendarsPromises` because the first one suggests there can be multiple destination calendars. It's ambiguous in that sense. I do understand that it's basically the same name as in the function (where it should be updated as well, ideally, for clarity, but that's beyond the scope of this PR)